### PR TITLE
doc: update docstrings in xmi.py

### PR DESCRIPTION
**Pull request by jdhughes-dev**

### CODE REVIEW

File: xmipy/xmi.py [hunk 1]
```diff
@@ -22,35 +22,100 @@ class Xmi(Bmi):
...
-        """ """
+        """Prepare a single time step.
+
+        Read data from input files and calculate the current time step length
+        and the simulation time at the end of the current time step.
+
+        Parameters
+        ----------
+        dt : float
+            MODFLOW time step length for the current time step. Currently
```
Comment (Hofer-Julian) [inline xmipy/xmi.py]: There shouldn't be MODFLOW specific docs within xmipy

File: xmipy/xmi.py [hunk 2]
```diff
@@ -22,35 +22,100 @@ class Xmi(Bmi):
...
         ...
 
     @abstractmethod
     def get_subcomponent_count(self) -> int:
-        """ """
+        """Get the number of Numerical Solutions in the simulation.
+
+        For most applications, this number will be equal to 1. Note that this
+        part of the XMI only works when the simulation is defined with a single
+        Solution Group
```
Comment (Hofer-Julian) [inline xmipy/xmi.py]: Also MODFLOW specific

File: xmipy/xmi.py [hunk 3]
```diff
@@ -22,35 +22,100 @@ class Xmi(Bmi):
...
+
+        Returns
+        -------
+        int
+          The number of MODFLOW 6 numerical solutions in the simulation.
```
Comment (Hofer-Julian) [inline xmipy/xmi.py]: MODFLOW specific

File: xmipy/xmi.py [hunk 4]
```diff
@@ -22,35 +22,100 @@ class Xmi(Bmi):
...
     def prepare_solve(self, component_id: int) -> None:
-        """ """
+        """Prepare for solving the system of equations.
+
+        This preparation mostly consists of advancing model states.
+
+        Parameters
+        ----------
+        component_id : int
+            MODFLOW 6 numerical solution id number.
```
Comment (Hofer-Julian) [inline xmipy/xmi.py]: MODFLOW specific

File: xmipy/xmi.py [hunk 5]
```diff
@@ -22,35 +22,100 @@ class Xmi(Bmi):
...
+
+        Formulate the system of equations for this outer (Picard) iteration and
+        solve. New data used when formulating the system of equations can be
+        injected prior to solve().  Before calling this, a matching call to
+        prepare_solve() should be performed.
+
+        Parameters
+        ----------
+        component_id : int
+            MODFLOW 6 numerical solution id number.
```
Comment (Hofer-Julian) [inline xmipy/xmi.py]: MODFLOW specific

File: xmipy/xmi.py [hunk 6]
```diff
@@ -22,35 +22,100 @@ class Xmi(Bmi):
...
+        """Finalize the model solve for the current time step.
+
+        Finalize model variables prior to calling finalize_time_step(). This
+        method should always be called after calls to prepare_solve() and
+        solve()
+
+        Parameters
+        ----------
+        component_id : int
+            MODFLOW 6 numerical solution id number.
```
Comment (Hofer-Julian) [inline xmipy/xmi.py]: MODFLOW specific

**Review by Hofer-Julian — CHANGES_REQUESTED** [review]

File: xmipy/xmi.py (hunk 2 above)
Comment (jdhughes-dev) [reply xmipy/xmi.py]: modified to be generic

File: xmipy/xmi.py (hunk 3 above)
Comment (jdhughes-dev) [reply xmipy/xmi.py]: modified to be generic

File: xmipy/xmi.py (hunk 4 above)
Comment (jdhughes-dev) [reply xmipy/xmi.py]: modified to be generic

File: xmipy/xmi.py (hunk 5 above)
Comment (jdhughes-dev) [reply xmipy/xmi.py]: modified to be generic

File: xmipy/xmi.py (hunk 6 above)
Comment (jdhughes-dev) [reply xmipy/xmi.py]: modified to be generic

File: xmipy/xmi.py (hunk 1 above)
Comment (jdhughes-dev) [reply xmipy/xmi.py]: modified to be generic

**Review by Hofer-Julian — APPROVED** [review]
Thanks!

**Pull request merged**